zilencer/views.py

```diff
@@ -532,6 +530,45 @@ def update_remote_realm_data_for_server(
     except IntegrityError:
         raise JsonableError(_("Duplicate registration detected."))
 
+    uuid_to_realm_dict = {str(realm["uuid"]): realm for realm in server_realms_info}
+    remote_realms_to_update = []
+    remote_realm_audit_logs = []
+    now = timezone_now()
+
+    # Update RemoteRealm entries, for which the corresponding realm's info has changed
+    # (for the attributes that make sense to sync like this).
+    for remote_realm in already_registered_remote_realms:
+        modified = False
+        realm = uuid_to_realm_dict[str(remote_realm.uuid)]
+        for remote_realm_attr, realm_dict_key in [
+            ("host", "host"),
+            ("realm_deactivated", "deactivated"),
+        ]:
+            old_value = getattr(remote_realm, remote_realm_attr)
+            new_value = realm[realm_dict_key]
+            if old_value != new_value:
+                setattr(remote_realm, remote_realm_attr, new_value)
+                remote_realm_audit_logs.append(
+                    RemoteRealmAuditLog(
+                        server=server,
+                        remote_id=None,
+                        realm_id=realm["id"],
+                        event_type=RemoteRealmAuditLog.REMOTE_REALM_VALUE_UPDATED,
+                        event_time=now,
+                        extra_data={
+                            "attr_name": remote_realm_attr,
+                            "old_value": old_value,
+                            "new_value": new_value,
+                        },
+                    )
+                )
+                modified = True
+        if modified:
+            remote_realms_to_update.append(remote_realm)
```

Comment:
This is pretty ugly, happy to rewrite this if there's a nice pattern. Would look a bit better if we didn't the dispary between `RemoteRealm.realm_deactivated` vs `Realm.deactivated` name that requires saving a mapping, but stilll...

Also, thoughts on whether `host` and `realm_deactivated` are the right values to sync?
- uuid doesn't make sense to sync, since it's the identifier
- uuid_owner_secret is an auth credential, so probably shouldn't be subject to auto-sync like this?
- date_created should be constant, so I don't see a use case for syncing it aside of tampering with our record in this table.

Reply:
I think this is great! We can always tweak from here in the future. The main change I'd make is using the early-continue pattern; will do that.